test/ng/directive/ngOptionsSpec.js

```diff
@@ -922,6 +922,25 @@ describe('ngOptions', function() {
       expect(scope.selected).toEqual([20]);
       expect(element).toEqualSelectValue([20], true);
     });
+
+
+    it('should ignore array values beginning with $$', function() {
+      scope.arr = ['one', 'two', '$$hashKey'];
```

Comment:
I'm not sure why it makes sense to ignore array elements which are strings beginning with "$$", that doesn't communicate "private" to me